src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -38,9 +38,10 @@ internal class SubstitutedMethodSymbol : WrappedMethodSymbol
 
         private int _hashCode; // computed on demand
 
-        internal SubstitutedMethodSymbol(SubstitutedNamedTypeSymbol containingSymbol, MethodSymbol originalDefinition)
+        internal SubstitutedMethodSymbol(NamedTypeSymbol containingSymbol, MethodSymbol originalDefinition)
             : this(containingSymbol, containingSymbol.TypeSubstitution, originalDefinition, constructedFrom: null)
         {
+            Debug.Assert(containingSymbol is SubstitutedNamedTypeSymbol || containingSymbol is SubstitutedErrorTypeSymbol);
```

Comment:
> Error symbols are designed specifically to be "used as if they are good" to avoid having to special-case error recovery code.

It is very possible that that was the intent behind introducing non-type error symbols. However, the implementation never delivered on that promise. Until that is done, I think we should move away from the pattern of using "Unsafe..." helpers in the local rewriter to using "TryGet..." helpers. PR https://github.com/dotnet/roslyn/pull/17549 made local rewriter more robust in situations like this and I believe we should follow the same approach in addressing this particular issue. Judging by the call stack, the crash that we are trying to fix happens in UnsafeGetNullableMethod. It is already marked as an unsafe API to use.